### PR TITLE
[Performance-Integration-test-ImportAndDeleteObjectEntryPerformanceTest]LPD-37359 Create performance integration test for Object deletion sce…

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPD-37359

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Lily Chi
+ */
+@DataGuard(scope = DataGuard.Scope.NONE)
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:44]: Any reason to set data guard to none?

Comment (slnn) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:44]: Follow the same rule in UserAccountResourcePerformanceTest

**Comment by dantewang** [comment]
ci:test:relevant

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new AssumeTestRule("assume"), new LiferayIntegrationTestRule());
+
+	public static void assume() {
+		Assume.assumeTrue(Validator.isNull(System.getenv("JENKINS_HOME")));
+	}
+
+	@BeforeClass
+	public static void setUpClass() throws Exception {
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:59]: @slnn If your test contains only 1 test method, and you don't have any plan to add more, you may not need `setUpClass()` and `setUp()`. Just put all the logic in the test method.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+			"/object-folder-definition.json");
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_importObjectFolder();
+	}
+
+	@Test
+	public void testDeleteObjectEntry() throws Exception {
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:71]: @slnn Both import and delete are time-logged. The method should not be called `testDelete...`.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+			httpInvoker.invoke();
+
+			while (_currentObjectEntryCount != 0) {
+				_currentObjectEntryCount =
+					_objectEntryLocalService.getObjectEntriesCount(
+						_objectDefinition.getObjectDefinitionId());
+			}
+		}
+		catch (Exception exception) {
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:112]: @slnn Don't catch and rethrow `RuntimeException`. Just make the methods `throws Exception`.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+		for (int i = 0; i < _OBJECT_ENTRY_COUNT; i++) {
+			jsonArray.put(JSONUtil.put("alpha", "foo"));
+		}
+
+		return jsonArray.toString();
+	}
+
+	private void _deleteObjectEntry() {
+		try (LoggingTimer loggingTimer = new LoggingTimer(
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:92]: @slnn Brian may reject this as it's called `PerformanceTest`.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+			HttpInvoker httpInvoker = HttpInvoker.newHttpInvoker();
+
+			httpInvoker.body(jsonArray.toString(), "application/json");
+			httpInvoker.userNameAndPassword(_USER_NAME_AND_PASSWORD);
+			httpInvoker.httpMethod(HttpInvoker.HttpMethod.DELETE);
+			httpInvoker.path(_PATH);
+
+			httpInvoker.invoke();
+
+			while (_currentObjectEntryCount != 0) {
```
Comment (dantewang) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ImportAndDeleteObjectEntryPerformanceTest.java:106]: @slnn Please make `_currentObjectEntryCount` local variables, not a field. For example, when you delete, you know how many entries needs to be deleted, so you shouldn't rely on the count resulted from the import counting.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by slnn** [comment]
Please see https://github.com/liferay-core-infra/liferay-portal/pull/7614

**Pull request closed without merging**